html/controllers/sign_up.php

```diff
@@ -0,0 +1,78 @@
+<?php
+// 新規登録機能
+function signUp($db)
+{
+    // jsonを取得
+    $json = file_get_contents("php://input");
+    $params = json_decode($json, true)['sign_up_user_params'];
+    $name = $params['name'];
+    $bio = $params['bio'];
+    $email = $params['email'];
+    $password = $params['password'];
+    $passwordConfirm = $params['password_confirmation'];
+
+    // token生成
+    $salt = "phpapi";
+    $seed = $salt . $email;
+    $token = hash('sha256', $seed);
+
+    // blankチェック
+    if ($name === "") {
+        $errorMessage = "Validation failed: Name can't be blank";
+        sendResponse($errorMessage);
+    } elseif ($bio === "") {
+        $errorMessage = "Validation failed: Bio can't be blank";
+        sendResponse($errorMessage);
+    } elseif ($email === "") {
+        $errorMessage = "Validation failed: Email can't be blank";
+        sendResponse($errorMessage);
+    } elseif ($password === "") {
+        $errorMessage = "Validation failed: Password can't be blank";
+        sendResponse($errorMessage);
+
+    // password一致チェック
+    } elseif ($password != $passwordConfirm) {
+        $errorMessage = "Validation failed: Password confirmation doesn't match password";
+        sendResponse($errorMessage);
+    }
+
+    // email重複チェック
+    $selectUserByEmail = $db->prepare("SELECT * FROM users WHERE email = :email");
```

Comment:
シングルトンのpdoを使うときはこう書く。
この関数に引数でpdoのインスタンスを渡す必要はもうない。

```suggestion
    $selectUserByEmail = Db::getPdo()->prepare("SELECT * FROM users WHERE email = :email");
```